third_party/move/move-compiler-v2/tests/checking/receiver/bad_receiver.exp

```diff
@@ -0,0 +1,31 @@
+
+Diagnostics:
+error: undeclared receiver function `borrow` for type `vector<Entry<K, V>>`
+   ┌─ tests/checking/receiver/bad_receiver.move:25:10
+   │
+25 │         &map.entries.borrow(self.index).key
+   │          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: undeclared receiver function `lower_bound` for type `OrderedMap<K, V>`
+   ┌─ tests/checking/receiver/bad_receiver.move:29:27
+   │
+29 │         let lower_bound = self.lower_bound(key);
+   │                           ^^^^^^^^^^^^^^^^^^^^^
+
+error: type cannot have type arguments
+   ┌─ tests/checking/receiver/bad_receiver.move:39:36
+   │
+39 │     public fun borrow<K, V>(self: &OrderedMao<K, V>, key: &K): &V {
+   │                                    ^^^^^^^^^^
+
+error: undeclared `0x1::ordered_map::OrderedMao`
+   ┌─ tests/checking/receiver/bad_receiver.move:39:36
+   │
+39 │     public fun borrow<K, V>(self: &OrderedMao<K, V>, key: &K): &V {
+   │                                    ^^^^^^^^^^
+
+error: unable to infer instantiation of type `fun self.iter_borrow(Self,&*error*):&V` (consider providing type arguments or annotating the type)
```

Comment:
The type `fun self.iter_borrow(Self,&*error*):&V` mentioned here is not particularly helpful to the user. Maybe we could display something better when `*error*` is involved?

Could be addressed in a separate PR though.

Reply:
The `*error*` shown here is the type `K` in the parameter type.  What we probably should do is to not print this error message, since we already see an`*error*` type and we presumably print another error message when generating that type.